### PR TITLE
generalize mpe

### PR DESCRIPTION
**Pull request by Tinkaa**
moving the function to `utils_general` and taking out the mean calculation such that it can be used more generally.

### CODE REVIEW

File: src/utils_general/statistics.py [hunk 1]
```diff
@@ -92,3 +92,12 @@ def get_return_period_function_empirical(df_rp: pd.DataFrame, rp_var: str):
     df_rp["exceedance_probability"] = df_rp["rank"] / (n + 1)
     df_rp["rp"] = 1 / df_rp["exceedance_probability"]
     return interp1d(df_rp["rp"], df_rp[rp_var])
+
+
+def calc_mpe(observations: np.array, forecast: np.array) -> float:
+    denominator = observations
```
Comment (turnerm) [inline src/utils_general/statistics.py:98]: I think this line can be removed and `observations` can be added directly as the denominator.

File: src/utils_general/statistics.py [hunk 2]
```diff
@@ -92,3 +92,12 @@ def get_return_period_function_empirical(df_rp: pd.DataFrame, rp_var: str):
     df_rp["exceedance_probability"] = df_rp["rank"] / (n + 1)
     df_rp["rp"] = 1 / df_rp["exceedance_probability"]
     return interp1d(df_rp["rp"], df_rp[rp_var])
+
+
+def calc_mpe(observations: np.array, forecast: np.array) -> float:
```
Comment (turnerm) [inline src/utils_general/statistics.py:97]: Maybe you can add a docstring, like in the functions above?

Comment (turnerm) [reply src/utils_general/statistics.py:97]: (and btw the other docstring formats are a bit messed up from the Black / flake8 formatting, and I didn't catch it in the PR ><. Each `param` should be on a new line)

File: src/indicators/flooding/glofas/utils.py [hunk 3]
```diff
@@ -324,16 +324,6 @@ def get_rank(observations: np.array, forecast: np.array) -> np.array:
     return rank
 
 
-def calc_mpe(observations: np.array, forecast: np.array) -> float:
```
Comment (turnerm) [inline src/indicators/flooding/glofas/utils.py:327]: Removing this completely will break some GloFAS notebooks. Do you think that maybe you could add the method back, and have it call the one from `utils_general.statistics`?

File: analyses/npl/02_glofas_skill.md [hunk 4]
```diff
@@ -164,9 +165,11 @@ for basin, stations in STATIONS_BY_MAJOR_BASIN.items():
         mpe_ev = np.empty(len(da_forecast.leadtime))
         for ilt, leadtime in enumerate(da_forecast.leadtime):
             observations, forecast = utils.get_same_obs_and_forecast(da_observations, da_forecast, leadtime)
-            mpe[ilt] = utils.calc_mpe(observations, forecast)
+            mean_forecast = forecast.mean(axis=0)
+            mpe[ilt] = calc_mpe(observations, mean_forecast)
```
Comment (turnerm) [inline analyses/npl/02_glofas_skill.md:169]: Thanks for updating this!

**Review by turnerm — APPROVED** [review]

File: src/utils_general/statistics.py (hunk 2 above)
Comment (Tinkaa) [reply src/utils_general/statistics.py:97]: Oh no! good catch, apparently the tool I used to wrap them messed this up. Fixed them here, but didn't feel like fixing all of them 😅

**Pull request merged**